File: java/org.financer.client/src/main/java/org/financer/client/connection/RestCallback.java
package org.financer.client.connection;

import org.financer.shared.connection.AsyncCall;

public interface RestCallback<T> extends AsyncCall<T> {

    @Override
    void onSuccess(T result);

    @Override
    default void onFailure(Exception exception) {
    }

    default void onBefore() {
    }

    @Override
    default void onAfter() {
    }

}

File: java/org.financer.client/src/main/kotlin/org/financer/client/connection/RestCallback.kt
package org.financer.client.connection

import org.financer.shared.connection.AsyncCall

interface RestCallback<T> : AsyncCall<T> {
    override fun onSuccess(result: T)
    override fun onFailure(exception: Exception) {}
    fun onBefore() {}
    override fun onAfter() {}
}